### PR TITLE
Fixes a regularization error that caused exploding gradients in SI-SDR loss.

### PR DESCRIPTION
**Pull request by pseeth**
Also cleaned up one of the PIT test cases.

### CODE REVIEW

**Comment by ethman** [comment]
Can you explain the issue with the SI-SNR loss? Was the epsilon being too big really the problem? Also what's going on with the dims now?

**Comment by pseeth** [comment]
The key was the `eps` in the `torch.log10` call, I believe. In any case it stabilized training. PIT also was reducing across both dimensions in the inner loop which was fine but unnecessary, due to the `torch.min` later in the code.

**Comment by ethman** [comment]
Yeah, okay. Makes sense, I was confused by how the dimensions fit into the exploding gradient.

While you're in this loss function, can you add an `SNR_max` like in the MixIT papers? Scott Wisdom mentioned that clipping the gradient above some dB threshold prevented the net from ignoring poorer performing examples. Reference: https://openreview.net/pdf?id=qMMzJGRPT2d, and https://arxiv.org/pdf/2006.12701.pdf

**Comment by pseeth** [comment]
The test appeared to be wrong (was comparing output of two different loss dictionaries that were not guaranteed to be the same). Fixed it - I think it was only incidentally passing before. The PIT loss change is definitely fine though.

**Comment by ethman** [comment]
LGTM.
